src/lib/index.ts

```diff
@@ -0,0 +1,272 @@
+      return now >= this.expiresAt;
+    } else {
+      return true;
+    }
+  }
+
+  /**
+   * Returns a cached token or retrieves a new one from Google.
+   *
+   * @param callback The callback function.
+   */
+  public getToken(callback: (err: Error|null, token?: string|null) => void) {
+    const handleJSONKey = (err: Error, key: string) => {
+      if (err) {
+        callback(err);
+        return;
+      }
+      try {
+        const body = JSON.parse(key);
+        this.key = body.private_key;
+        this.iss = body.client_email;
+      } catch (e) {
+        callback(e);
+        return;
+      }
+
+      if (!this.key || !this.iss) {
+        const error = new Error('private_key and client_email are required.');
+        (<NodeJS.ErrnoException>error).code = 'MISSING_CREDENTIALS';
+        callback(error);
+        return;
+      }
+
+      this.requestToken(callback);
+    };
+
+    const handleKey = (err: Error, key: string) => {
+      if (err) {
+        callback(err);
+        return;
+      }
+      this.key = key;
+      this.requestToken(callback);
+    };
+
+    if (!this.hasExpired()) {
+      return callback(null, this.token);
+    } else {
+      if (!this.key && !this.keyFile) {
+        callback(new Error('No key or keyFile set.'));
+        return;
+      } else if (!this.key && this.keyFile) {
+        const mimeType = mime.getType(this.keyFile);
+        if (mimeType === 'application/json') {
+          // json file
+          fs.readFile(this.keyFile, 'utf8', handleJSONKey);
+        } else {
+          // Must be a .p12 file or .pem key
+          if (!this.iss) {
+            const error = new Error('email is required.');
+            (<NodeJS.ErrnoException>error).code = 'MISSING_CREDENTIALS';
+            callback(error);
+            return;
+          }
+
+          if (mimeType === 'application/x-pkcs12') {
+            // convert to .pem on the fly
+            gp12pem(this.keyFile, handleKey);
+          } else {
+            // assume .pem key otherwise
+            fs.readFile(this.keyFile, 'utf8', handleKey);
+          }
+        }
+      } else {
+        return this.requestToken(callback);
+      }
+    }
+  }
+
+  /**
+   * Revoke the token if one is set.
+   *
+   * @param callback The callback function.
+   */
+  public revokeToken(callback: (err?: Error) => void) {
+    if (this.token) {
+      request(GOOGLE_REVOKE_TOKEN_URL + this.token, (err: Error) => {
+        if (err) {
+          callback(err);
+          return;
+        }
+        this.configure({
+          email: this.iss,
+          sub: this.sub,
+          key: this.key,
+          keyFile: this.keyFile,
+          scope: this.scope
+        });
+        callback();
+      });
+    } else {
+      callback(new Error('No token to revoke.'));
+    }
+  }
+
+  /**
+   * Configure the GoogleToken for re-use.
+   * @param  {object} options Configuration object.
+   */
+  private configure(options?: TokenOptions) {
```

Comment:
Use default argument: `configure(options: TokenOptions = {})` and then nuke the next line.

Reply:
fixed